src/main/java/bio/terra/common/UriUtils.java

```diff
@@ -0,0 +1,70 @@
+package bio.terra.common;
+
+import autovalue.shaded.kotlin.text.Charsets;
+import bio.terra.service.filedata.exception.InvalidUriException;
+import java.net.URI;
+import java.net.URISyntaxException;
+import java.util.List;
+import java.util.Objects;
+import org.apache.http.NameValuePair;
+import org.apache.http.client.utils.URIBuilder;
+import org.apache.http.client.utils.URLEncodedUtils;
+
+/** Handy methods for working with URIs */
+public class UriUtils {
+
+  private UriUtils() {}
+
+  /**
+   * Extract the value from a query parameter
+   *
+   * @param uri a valid uri
+   * @param parameter the name of the parameter whose value to read
+   * @return the value of the query parameter or null. If there are multiple instances of the query
+   *     parameter, return the value as a comma-separated list
+   */
+  public static String getValueFromQueryParameter(String uri, String parameter) {
+    List<String> values =
+        URLEncodedUtils.parse(toUri(uri), Charsets.UTF_8).stream()
+            .filter(p -> Objects.equals(p.getName(), parameter))
+            .map(NameValuePair::getValue)
+            .toList();
+    // Explicitly return null if there were no matching query parameters
+    if (values.isEmpty()) {
+      return null;
+    }
+    return String.join(",", values);
+  }
+
+  /**
+   * Rewrite the uri with the specified query parameter removed. Note: the order of any existing
+   * parameters may not be maintained
+   *
+   * @param uri the uri to rewrite
+   * @param parameter the parameter to exclude
+   * @return a new uri string any reference to the query parameter removed
+   */
+  public static String omitQueryParameter(String uri, String parameter) {
+    URI initUri = toUri(uri);
+    List<NameValuePair> newQueryParams =
+        URLEncodedUtils.parse(toUri(uri), Charsets.UTF_8).stream()
+            .filter(p -> !Objects.equals(p.getName(), parameter))
+            .toList();
+    return new URIBuilder(initUri).setParameters(newQueryParams).toString();
```

Comment:
Have you considered using spring `UriComponents` for this kind of operation? for omitting a query parameter, something like this should work:
```
    return UriComponentsBuilder.fromUriString(uri).replaceQueryParam(parameter).build().toString();
```

To get a query param, you'd write something like:
```
  List<String> value = UriComponentsBuilder.fromUriString(uri).build().getQueryParams().get(parameter);
```

Reply:
ah sadness, I reverted it because it breaks URIs in certain cases (and I don't really care that much about drilling in...I don't really trust this method anymore)

Reply:
actually, nevermind...might be something I did :(